cla_backend/apps/cla_eventlog/tests/test_find_and_delete_old_cases.py

```diff
@@ -0,0 +1,192 @@
+from django.test import TestCase
+from django.utils import timezone
+from freezegun import freeze_time
+import datetime
+
+from core.tests.mommy_utils import make_recipe
+from cla_eventlog.models import Log
+from complaints.models import Complaint
+from legalaid.models import Case, EODDetails, EligibilityCheck, PersonalDetails
+from diagnosis.models import DiagnosisTraversal
+from cla_auditlog.models import AuditLog
+from cla_eventlog.management.commands.find_and_delete_old_cases import FindAndDeleteCasesUsingCreationTime
+
+
+def _make_datetime(year=None, month=None, day=None, hour=0, minute=0, second=0):
+    today = datetime.date.today()
+    year = year if year else today.year
+    month = month if month else today.month
+    day = day if day else today.day
+    dt = datetime.datetime(year, month, day, hour, minute, second)
+    return timezone.make_aware(dt, timezone.get_current_timezone())
+
+
+class FindAndDeleteOldCases(TestCase):
+    def setUp(self):
+        super(FindAndDeleteOldCases, self).setUp()
+        self.instance = FindAndDeleteCasesUsingCreationTime()
+
+    def create_case(self, current_time, case_type="legalaid.case"):
+        case = None
+        freezer = freeze_time(current_time)
+        freezer.start()
+        case = make_recipe(case_type)
+        freezer.stop()
+        return case
+
+    def create_event_log_for_case(self, case, code, created):
+        make_recipe("cla_eventlog.log", case=case, code=code, created=created)
+
+    def delete_old_cases(self, current_time):
+        self.instance.run()
+
+    def find_old_cases(self, current_time):
+        cases = None
+        freezer = freeze_time(current_time)
+        freezer.start()
+        cases = self.instance.get_eligible_cases()
+        freezer.stop()
+        return cases
+
+    def test_old_case_with_recent_event_logs(self):
+        date = _make_datetime(year=2014, month=4, day=27, hour=9)
+        case = self.create_case(date)
+
+        self.create_event_log_for_case(case, "CASE_VIEWED", _make_datetime(year=2014, month=5, day=30, hour=9))
+        self.create_event_log_for_case(case, "CASE_VIEWED", _make_datetime(year=2018, month=4, day=27, hour=9))
+        self.create_event_log_for_case(case, "CASE_VIEWED", _make_datetime(year=2020, month=4, day=27, hour=9))
+
+        dt = _make_datetime(year=2021, month=1, day=1, hour=9)
+        oldCasesFound = self.find_old_cases(dt)
+        self.assertEqual(oldCasesFound.count(), 0)
+        # Only need to check for this once as every test case uses this method via calling self.delete_old_cases
+        with self.assertNumQueries(1):
```

Comment:
Given the changes you've made to the management command, I'd like to see an `assertNumQueries` for running the whole command in "find" mode, and not just callings its `get_eligible_cases` method directly

Reply:
Got it